docs/specification/draft/basic/authorization.mdx

```diff
@@ -63,119 +65,82 @@ audience. For instance:
 Server Metadata Discovery to allow the MCP server to respond based on the MCP protocol
 version.
 
-For example: `MCP-Protocol-Version: 2024-11-05`
-
-#### 2.3.2 Authorization Base URL
+MCP servers **SHOULD** use the `MCP-Protocol-Version` header to determine compatibility with the MCP client.
 
-The authorization base URL **MUST** be determined from the MCP server URL by discarding
-any existing `path` component. For example:
-
-If the MCP server URL is `https://api.example.com/v1/mcp`, then:
+For example: `MCP-Protocol-Version: 2024-11-05`
 
-- The authorization base URL is `https://api.example.com`
-- The metadata endpoint **MUST** be at
-  `https://api.example.com/.well-known/oauth-authorization-server`
+#### 2.3.4 Sequence Diagram
+The following diagram outlines an example flow:
 
-This ensures authorization endpoints are consistently located at the root level of the
-domain hosting the MCP server, regardless of any path components in the MCP server URL.
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant M as MCP Server (Resource Server)
+    participant A as Authorization Server
 
-#### 2.3.3 Fallbacks for Servers without Metadata Discovery
+    C->>M: MCP Request without token
+    M-->>C: HTTP 401 Unauthorized with WWW-Authenticate header
+    Note over C: Extract resource_metadata<br />from WWW-Authenticate
 
-For servers that do not implement OAuth 2.0 Authorization Server Metadata, clients
-**MUST** use the following default endpoint paths relative to the authorization base URL
-(as defined in [Section 2.3.2](#232-authorization-base-url)):
+    C->>M: GET /.well-known/oauth-protected-resource
+    M-->>C: Resource metadata with authorization server URL
+    Note over C: Validate RS Metadata,<br />build AS metadata URL
 
-| Endpoint               | Default Path | Description                          |
-| ---------------------- | ------------ | ------------------------------------ |
-| Authorization Endpoint | /authorize   | Used for authorization requests      |
-| Token Endpoint         | /token       | Used for token exchange & refresh    |
-| Registration Endpoint  | /register    | Used for dynamic client registration |
+    C->>A: GET /.well-known/oauth-authorization-server
+    A-->>C: Authorization server metadata
 
-For example, with an MCP server hosted at `https://api.example.com/v1/mcp`, the default
-endpoints would be:
+    Note over C,A: OAuth 2.1 authorization flow happens here
 
-- `https://api.example.com/authorize`
-- `https://api.example.com/token`
-- `https://api.example.com/register`
+    C->>A: Token Request
+    A-->>C: Access Token
 
-Clients **MUST** first attempt to discover endpoints via the metadata document before
-falling back to default paths. When using default paths, all other protocol requirements
-remain unchanged.
+    C->>M: MCP Request with Access Token
+    M-->>C: Protected Resource Response
```

Comment:
There is a mismatch between "MCP Request" followed by "Protected Resource Response". These should probably be made consistent, either "MCP Request"/"MCP Response" or "Protected Resource Request"/"Protected Resource Response"